Hec.Cwms/CwmsDatabase.cs

```diff
@@ -69,21 +69,38 @@ public string[] GetOfficeIDS()
 
     private string LookupUnits(string tsid)
     {
-      DataTable tbl = oracle.Table("cwms_v_ts_id", "select DB_OFFICE_ID, CWMS_TS_ID,UNIT_ID from cwms_v_ts_id where CWMS_TS_ID = '" + tsid + "' and DB_OFFICE_ID='" + officeID + "' ");
+      DataTable tbl = oracle.Table("cwms_v_ts_id", "select DB_OFFICE_ID, CWMS_TS_ID,UNIT_ID from cwms_v_ts_id where CWMS_TS_ID = '" + tsid + "' and DB_OFFICE_ID='" + officeID + "'");
       if (tbl.Rows.Count > 0)
-        return tbl.Rows[0]["UNIT_ID"].ToString();
+        return tbl.Rows[0][2].ToString();
       return "";
     }
-
+        /// <summary>
+        /// Uses a switch case take in metric untis and return english units
+        /// </summary>
+        /// <param name="units"></param>
+        /// <returns></returns>
+        private string ChangeMetricToEnglish(string units)
+        {
+            switch (units.ToLower())
+            {
+                case "cms":
+                    return "cfs";
+                case "m":
+                    return "feet";
+                case "mm":
+                    return "in";
+            }
+            return "";
+        }
 
     public TimeSeries ReadTimeSeries(string tsid, DateTime t1, DateTime t2)
```

Comment:
That extra trip to the Sever to lookup units would slow things down as this scales up. How about adding a optional units argument to the ReadTimeSeries(....., string units="").  And if unit is not blank , then avoid the extra trip to the database.

Reply:
And in the main program you could recycle the units read from excel.

Reply:
Mike gave me a tip on making one database trip for returning both the units and the data in English units. 
```
cwms_display.retrieve_user_unit_f(cwms_util.split_text(:tsid, 2, '.'), 'EN')
```

In our cwms_ts.retireve_ts we could replace `:units` with the above code. 

I'm going to merge in what I've got right now. Then i'll implement the new retrieve in a new branch.